### PR TITLE
Refactors SpotLapseRateAdjust so that the two pathways are easier to follow.

### PR DESCRIPTION
**Pull request by MoseleyS**
Following https://github.com/metoppv/improver/pull/1879, I want to refactor this code so that it is easier to see what is going on.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: improver/spotdata/apply_lapse_rate.py [hunk 1]
```diff
@@ -211,32 +161,100 @@ def process(
...
+                A cube of temperature lapse rates on the same grid as that from
+                which the spot data temperatures were extracted. If not provided
+                a fixed lapse rate must have been specified.
+
+        Returns:
+            A copy of the input spot_data_cube with the data modified by
+            the lapse rates to give a better representation of the site's
+            temperatures.
+
+        Raises:
```
Comment (bayliffe) [inline improver/spotdata/apply_lapse_rate.py:234]: Can you move the Raises sections to the `_check_inputs` doc string.

**Review by bayliffe — CHANGES_REQUESTED** [review]
Thanks, this looks better. Only a tiny comment to address.

**Review by bayliffe — DISMISSED** [review]

File: improver/spotdata/apply_lapse_rate.py [hunk 2]
```diff
@@ -80,66 +80,35 @@ def __init__(
...
         )
 
-    def process(
-        self,
-        spot_data_cube: Cube,
-        neighbour_cube: Cube,
-        gridded_lapse_rate_cube: Cube = None,
-    ) -> Cube:
+    def _check_inputs(
+        self, gridded_lapse_rate_cube: Cube, neighbour_cube: Cube, spot_data_cube: Cube
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/spotdata/apply_lapse_rate.py:92]: Done

File: improver/spotdata/apply_lapse_rate.py [hunk 3]
```diff
@@ -211,32 +177,89 @@ def process(
...
-        if not use_fixed_lr:
-            # Extract the lapse rates that correspond to the spot sites.
-            spot_lapse_rate = SpotExtraction(
-                neighbour_selection_method=self.neighbour_selection_method
-            )(neighbour_cube, gridded_lapse_rate_cube)
+    def get_gridded_lapse_rate(
+        self,
+        spot_data_cube: Cube,
+        neighbour_cube: Cube,
+        gridded_lapse_rate_cube: Cube = None,
```
[the remark this answers is not in the document]
Comment (MoseleyS) [reply improver/spotdata/apply_lapse_rate.py:184]: I keep forgetting this. Added.

**Review by gavinevans — APPROVED** [review]

**Pull request merged**